lib/android/app/src/main/java/com/reactnativenavigation/views/TopBar.java

```diff
@@ -1,24 +1,32 @@
 package com.reactnativenavigation.views;
 
-import android.app.Activity;
 import android.content.Context;
 import android.graphics.Typeface;
 import android.support.annotation.ColorInt;
 import android.support.annotation.Nullable;
 import android.support.design.widget.AppBarLayout;
 import android.support.v7.widget.Toolbar;
+import android.view.Menu;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.TextView;
 
+import com.reactnativenavigation.parse.Button;
+import com.reactnativenavigation.viewcontrollers.ContainerViewController;
+
+import java.util.ArrayList;
+
 public class TopBar extends AppBarLayout {
 	private final Toolbar titleBar;
+	private ContainerViewController.ContainerView containerView;
 
-	public TopBar(final Context context) {
+	public TopBar(final Context context, ContainerViewController.ContainerView containerView) {
 		super(context);
-		titleBar = new Toolbar(context);
+
+		this.titleBar = new Toolbar(context);
+		this.containerView = containerView;
```

Comment:
why do you need it here?

Reply:
containerView has the emitter that is needed to send the event to React-Native

Reply:
yeah, i got it. It's ok :)